client/ayon_resolve/plugins/publish/collect_editorial_package.py

```diff
@@ -1,88 +1,51 @@
 
     order = pyblish.api.CollectorOrder - 0.49
     label = "Collect Editorial Package Instances"
     families = ["editorial_pkg"]
 
-    def process(self, context):
-        project_name = context.data["projectName"]
+    def process(self, instance):
+        project_name = instance.context.data["projectName"]
         self.log.info(f"project: {project_name}")
 
-        for media_pool_item in lib.iter_all_media_pool_clips():
-
-            data = media_pool_item.GetMetadata(lib.pype_tag_name)
-            if not data:
-                continue
-
-            try:
-                data = json.loads(data)
-            except json.JSONDecodeError:
-                self.log.warning(
-                    f"Failed to parse json data from media pool item: "
-                    f"{media_pool_item.GetName()}"
-                )
-                continue
-
-            # exclude all which are not productType editorial_pkg
-            if (
-                data.get("publish")
-                and data["publish"].get("productType") != "editorial_pkg"
-            ):
-                continue
-
-            instance = context.create_instance(name=media_pool_item.GetName())
+        media_pool_item = instance.data["transientData"]["timeline_pool_item"]
 
-            publish_data = data["publish"]
+        # get version from publish data and rise it one up
+        version = instance.data.get("version")
+        if version is not None:
+            version += 1
 
-            # get version from publish data and rise it one up
-            version = publish_data.get("version")
-            if version is not None:
-                version += 1
-
-                # make sure last version of product is higher than current
-                # expected current version from publish data
-                folder_entity = ayon_api.get_folder_by_path(
-                    project_name=project_name,
-                    folder_path=publish_data["folderPath"],
-                )
-                last_version = ayon_api.get_last_version_by_product_name(
-                    project_name=project_name,
-                    product_name=publish_data["productName"],
-                    folder_id=folder_entity["id"],
-                )
-                if last_version is not None:
-                    last_version = int(last_version["version"])
-                    if version <= last_version:
-                        version = last_version + 1
-
-                publish_data["version"] = version
-
-            publish_data.update(
-                {
-                    "mediaPoolItem": media_pool_item,
-                    "item": media_pool_item,
-                }
+            # make sure last version of product is higher than current
+            # expected current version from publish data
+            folder_entity = ayon_api.get_folder_by_path(
+                project_name=project_name,
+                folder_path=instance.data["folderPath"],
+            )
```

Comment:
Isn't this just `instance.data["folderEntity"]` ?

Reply:
I couldn't find `folderEntity` in neither `instance.data` nor `instance.context.data`, so leave it as it is for now.

Reply:
You are right - the folder entity gets collected here: https://github.com/ynput/ayon-core/blob/ab24518f96ecaed74593f32e10448122105b73e7/client/ayon_core/plugins/publish/collect_anatomy_instance_data.py#L40 which is at a later `order`